include/GafferSceneUI/ContextAlgo.h

```diff
@@ -74,6 +74,10 @@ void expand( Gaffer::Context *context, const GafferScene::PathMatcher &paths, bo
 /// Returns a new PathMatcher containing the new leafs of this expansion.
 GafferScene::PathMatcher expandDescendants( Gaffer::Context *context, const GafferScene::PathMatcher &paths, const GafferScene::ScenePlug *scene, int depth = Imath::limits<int>::max() );
 
+/// Removes paths from the current expansion.
+/// Returns a new PathMatcher containing the removed paths.
+GafferScene::PathMatcher collapse( Gaffer::Context *context, const GafferScene::PathMatcher &paths );
```

Comment:
No, I don't need it. I included it just to be able to re-implement `SceneView::collapseSelection()`, and I'm not sure its right (though I can't tell if its wrong through tests or using the gui).

If we ditch it, we do lose the "collapse in place" bit that we had originally. That was lose [here](https://github.com/GafferHQ/gaffer/pull/2084/commits/e407e3d7aec1f80e22ca3271906a9cd5cf986c9f#diff-19bd03836e3aea327f0e741396ffcf66L1423) and regained [here](https://github.com/GafferHQ/gaffer/commit/7fb2d9e8ef9523a5f777b89ada21041564b10665#diff-19bd03836e3aea327f0e741396ffcf66R1359).